Lib/fontTools/subset/__init__.py

```diff
@@ -2407,12 +2407,27 @@ def prune_post_subset(self, options):
             for subr in subrs.items:
                 subr.subset_subroutines (local_subrs, font.GlobalSubrs)
 
+        # Delete local SubrsIndex if empty
+        if hasattr(font, 'FDSelect'):
+            for fd in font.FDArray:
+                _delete_empty_subrs(fd)
+        else:
+            _delete_empty_subrs(font)
+
         # Cleanup
         for subrs in all_subrs:
             del subrs._used, subrs._old_bias, subrs._new_bias
 
     return True
 
+
+def _delete_empty_subrs(font):
+    if hasattr(font.Private, 'Subrs') and not font.Private.Subrs:
+        if 'Subrs' in font.Private.rawDict:
+            del font.Private.rawDict['Subrs']
```

Comment:
Why is this one needed?

Reply:
otherwise when compiling, the raw stuff gets written out

Reply:
I copied from a similar snippet in cffLib

Reply:
Yeah I see in the code.  If we were maintaining this properly we would have added a `__delattr__`. But not now.